xfer/http_publisher.go

```diff
@@ -0,0 +1,80 @@
+package xfer
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+
+	"github.com/weaveworks/scope/common/sanitize"
+)
+
+// HTTPPublisher publishes buffers by POST to a fixed endpoint.
+type HTTPPublisher struct {
+	url     string
+	token   string
+	probeID string
+}
+
+// NewHTTPPublisher returns an HTTPPublisher ready for use.
+func NewHTTPPublisher(target, token, probeID string) (string, *HTTPPublisher, error) {
+	targetAPI := sanitize.URL("http://", 0, "/api")(target)
+	resp, err := http.Get(targetAPI)
```

Comment:
This, OTOH, will block for upto 30 secs, for each host, on the resolver goroutine.